CodeGeneration/Sources/SyntaxSupport/SyntaxNodeKind.swift

```diff
@@ -336,8 +336,82 @@ public enum SyntaxNodeKind: String, CaseIterable, IdentifierConvertible, TypeCon
     }
   }
 
+  public var base: Self {
```

Comment:
I found it a bit difficult to understand why `Node.base` isn't another `Node` but a `SyntaxNodeKind`. For me, since `SyntaxNodeKind` represents the type, it becomes natural to describe the type hierarchy within `SyntaxNodeKind`, and `SyntaxNodeKind.base` pairs up quite nice with `SyntaxNodeKind.isBase`.

Maybe I could break down the multiple case statements to improve readability?